graylog2-server/src/main/java/org/graylog/plugins/views/search/engine/suggestions/SuggestionResponse.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.plugins.views.search.engine.suggestions;
+
+import com.google.auto.value.AutoValue;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+
+@AutoValue
+public abstract class SuggestionResponse {
+
+    public abstract String field();
+
+    public abstract String input();
+
+    public abstract List<SuggestionEntry> suggestions();
+
+    public abstract Optional<SuggestionError> suggestionError();
+
+
+    public static Builder builder(final String field, final String input) {
```

Comment:
I would suggest using specific factory methods instead of a builder when constructing instances. E.g. we could have:

```suggestion
    public static SuggestionResponse forSuggestions(final String field, final String input, final List<SuggestionEntry> suggestions) { ... }
    public static SuggestionResponse forError(final String field, final String input, final SuggestionError error) { ... }
```

We do not really need the flexibility of a builder here, because we have two different cases for creation. Modelling them like this makes it more obvious what the possible states could be.

Reply:
Make sense, done, thanks!